masonry/src/tree_arena.rs

```diff
@@ -181,15 +220,18 @@ impl<'a, Item> TreeArenaToken<'a, Item> {
     ///
     /// Returns a tuple of a shared reference to the child and a token to access
     /// its children.
-    pub fn get_child(&self, id: u64) -> Option<(&'_ Item, TreeArenaToken<'_, Item>)> {
+    pub fn get_child(&self, id: u64) -> Option<ArenaRef<'_, Item>> {
         for child in self.children {
             if child.id == id {
-                return Some((
-                    &child.item,
-                    TreeArenaToken {
+                return Some(ArenaRef {
+                    parent_id: self.id,
+                    id: Some(child.id),
+                    item: &child.item,
+                    children: ArenaRefChildren {
+                        id: Some(child.id),
                         children: &child.children,
                     },
-                ));
+                });
```

Comment:
I think a lot of that duplicated boilerplate could be avoided (same for `ArenaMut`)?

Reply:
I'm not worried about duplicated boilerplate in library code. Especially for TreeArena, the error surface isn't that large; since the rest of the codebase uses it everywhere, it has extensive test coverage.

Reply:
It's less about that it's well tested or errornous, but just IMO unnecessary clutter in code that may be simplified by containing it in a well named function, to focus on the more important bits?